gems/aws-sdk-s3/lib/aws-sdk-s3/file_downloader.rb

```diff
@@ -31,6 +31,10 @@ def download(destination, options = {})
           key: options[:key],
         }
         @params[:version_id] = options[:version_id] if options[:version_id]
+        @params[:checksum_mode] = options[:checksum_mode] || 'ENABLED'
```

Comment:
I think this line has the side effect of making every request from `FileDownloader` default to enabling the checksum_mode=ENABLED feature at the client/api level, which was previously not enabled, even if the new feature here (on_checksum_validated callbacks) is not being used.

Was that an intentional change here?

It also doesn't seem possible to disable this, because setting this to falsey values (nil, false, etc) will have it get set by this line; and trying to set it to something like `DISABLED` or `FALSE` will throw up an error from elsewhere, because `ENABLED` is the only valid value at the api level.

Reply:
This was an intentional change here to set the default to ENABLED as a best practice.  Checksums will only be validated when they exist on the object (if the object was uploaded without checksums, the download will succeed and nothing will be verified).

However - you are right - given the lack of support on the API for a `disabled` or `false` value, its impossible to disable this.  I'll release a patch today.